tm2/pkg/sdk/baseapp.go

```diff
@@ -652,11 +652,9 @@ func (app *BaseApp) runMsgs(ctx Context, msgs []Msg, mode RunTxMode) (result Res
 		// Each message result's Data must be length prefixed in order to separate
 		// each result.
 		data = append(data, msgResult.Data...)
-		events = append(events, msgResult.Events...)
-		defer func() {
```

Comment:
I think it was a good approach, but you likely shadowed a variable or overrode one that was already in use. It appears more like a variable usage race.

Reply:
2ace123c597ad4637fed9a1da7d101e1802b1b09

event from @notJoon #1653 is in context level unlike `msgResult.event` is defined for each msgs. So based on my understanding we don't need handle this in loop.